### PR TITLE
test(agent-config): improve coverage for agent_config and message_for…

### LINKED ISSUE #646: test: improve coverage for agent_config and message_formatter
## Context

PR #628 added tests for `extract_tool_trace` and `tool_trace_to_messages()`, but codecov flagged some uncovered paths in `agent_config.py` (87.5% patch coverage) and `serialization.py` (80%).

## What to do

Add tests covering:
- `agent_config.py`: the `extra == "allow"` branch, `reset_agent_config()` edge cases
- `serialization.py`: flow loading with agent blocks present (the `_agent_config_set = False` path)
- `message_formatter.py`: edge cases with malformed traces (missing `type` field, empty `contents`)

## Why

Bring patch coverage above the project threshold and prevent regressions in the new agent config infrastructure.

From PR #628 codecov feedback.

### PR DESCRIPTION
**Pull request by dhshah13**
## Summary

Closes #646

Adds 11 tests covering the uncovered code paths flagged by codecov on PR #628:

- agent_config.py : the `extra == "allow"` branch in `set_agent_config()`, sensitive param redaction within that branch, `reset_agent_config()` edge cases (no agent blocks, never configured)

- serialization.py: flow loading with agent blocks sets `_agent_config_set = False`, flow loading without agent blocks sets `_agent_config_set = True`

- message_formatter.py: malformed trace edge cases (missing `type` field, empty `contents` list, missing `header` field, missing `text` in content item, non-list `content` field)

## Test plan
- [x] All 11 new tests pass locally
- [x] Full test suite passes (797 tests, 0 failures)
- [x] ruff lint clean
- [x] ruff format clean
- [x] Pre-commit hooks pass (ruff, ruff-format, conventional-pre-commit)
- [x] No production code changes — tests only

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage for agent-configuration workflows: added scenarios for setting, resetting, and detecting agent config, YAML-loading edge cases, and verification that sensitive values are redacted from logs.
  * Added unit tests for message formatting to handle malformed tool traces, missing or empty fields, non-list or unexpected content types, and numeric or edge-case outputs.

### CODE REVIEW

File: tests/flow/test_agent_config.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                agent_url="http://new:7860",
+                secret="super-secret-value",
+            )
+
+        assert flow.blocks[0].secret == "super-secret-value"
+
+        # The secret value must not appear in any log message
+        all_log_text = " ".join(record.message for record in caplog.records)
+        assert "super-secret-value" not in all_log_text
+
```
[the remark this answers is not in the document]
Comment (dhshah13) [reply tests/flow/test_agent_config.py]: The test now asserts both:

The secret value is not in the logs
The redacted marker "secret set (redacted)" is in the logs

**Review by shivchander — COMMENTED** [review]
## PR Review Summary

**Scope:** Test-only, 263 additions across 2 files. No production code changes. Zero risk to existing functionality.

---

### Important Issues (3)

**1. `from_yaml` mock tests silently bypass block construction validation**
`tests/flow/test_agent_config.py:127-165` and `:167-205`

The `Mock()` class used for `BlockRegistry._get` silently accepts and ignores all constructor kwargs. The YAML config values (`block_name`, `input_cols`) are never validated against the block constructor. If `create_block_from_config` changes its pre-processing logic, the mock would mask the regression.

Recommendation — add mock call assertions:
```python
mock_get.assert_called_once_with("AgentBlock")
mock_block_class.assert_called_once()
```

**2. `extra=="allow"` test lacks explicit precondition assertion**
`tests/flow/test_agent_config.py:59-74`

The test relies on `MockBlock` inheriting `model_config = ConfigDict(extra="allow")` from `BaseBlock` but never asserts this. If `BaseBlock` changes to `extra="ignore"` or `extra="forbid"`, the test fails with a confusing `AttributeError` instead of a clear message.

Recommendation:
```python
assert agent_block.model_config.get("extra") == "allow", \
    "Precondition: block must allow extra fields for this branch"
```

**3. Helper function duplication with existing test patterns**
`tests/flow/test_agent_config.py:37-52`

`_create_mock_block()` and `_create_mock_agent_block()` duplicate logic from `conftest.py`'s `mock_block` fixture and `test_base.py`'s `create_mock_agent_block` class method. Three separate locations would need updating if `MockBlock` construction changes. Consider reusing the existing fixture or adopting the class-method pattern from `test_base.py`.

---

### Suggestions (5)

**4. Misleading `serialization._agent_config_set` notation in comments**
Module docstring (line 7) and "Gap 3" section header (line 56) say `serialization._agent_config_set`, which reads as if `_agent_config_set` is a module-level symbol. It's a `PrivateAttr` on the `Flow` class in `base.py`, set by `load_flow_from_yaml()`. Suggested rewording: `"from_yaml sets flow._agent_config_set"`.

**5. `test_empty_content_list` docstring is subtly misleading**
Docstring says "falls through to structuredContent" but the test dict has no `structuredContent` key — the fallback is `json.dumps(output)` (the whole dict). Suggested: `"empty content list is falsy, so the entire output dict is JSON-serialized"`.

**6. Missing companion test for unknown (but present) `type` value**
`test_missing_type_field` tests `type=None`, but no test covers `{"type": "image"}` or other unrecognized types. A companion test would guard against partial handling of new types.

**7. Missing test for non-dict/non-string/non-None tool output**
`_extract_tool_output` has a `json.dumps(output)` path for truthy non-dict/non-string values (e.g., `output=42`). This path is not covered.

**8. Mock agent block uses `setattr` rather than Pydantic field declarations**
`_create_mock_agent_block` sets agent attributes via `setattr`, making them Pydantic "extra" fields rather than declared fields. Sufficient for current `hasattr`-based checks but would diverge from real `AgentBlock` if code switches to `model_fields` introspection. A comment documenting this tradeoff would help.

---

### Strengths

- Every test addresses a specific uncovered branch with clear rationale — no boilerplate filler
- `test_set_agent_config_extra_allow_sensitive_redaction` guards against vacuous passes with a positive assertion (`"secret set (redacted)" in all_log_text`)
- Proper `caplog` scoping with correct logger name and DEBUG level
- Behavior-focused assertions check observable outcomes rather than implementation internals
- Edge case tests for `message_formatter` handle real production concerns (malformed traces from Langflow)
- "Gap N" section headers clearly tie each test group to its coverage motivation

---

**Overall:** Solid test-only PR that correctly fills the coverage gaps it claims. The important issues are about test resilience, not correctness. Recommend addressing issues 1-2 (low effort, high value) before merge.

---
🤖 Generated with [Claude Code](https://claude.ai/code) — Claude Opus 4.6 (1M context)

**Comment by dhshah13** [comment]
Sorry about this - ruff version is keep annoying me :P
